arch/x86/core/multiboot.c

```diff
@@ -11,6 +11,43 @@
 
 struct multiboot_info multiboot_info;
 
+#ifdef CONFIG_BOOTARGS
+
+static const char *multiboot_cmdline;
+
+const char *get_bootargs(void)
+{
+	return multiboot_cmdline;
+}
+
+/*
+ * We have to map cmdline from multiboot, but we have to know its size
+ * to do it. Its not provided as a separate field, so we have to calculate
+ * it here by mapping it part by part.
```

Comment:
Similar comments, though I'm much less an expert on multiboot.   This seems way too complicated, trying to extract the string from "foreign" unmapped memory after OS entry.   Multi boot enters the OS in 32 bit mode with paging disabled, doesn't it?  Wouldn't it make much more sense to copy the string up in the stub than it does here?